pages/index.tsx

```diff
@@ -1,63 +1,179 @@
+    };
+
+    setUploadAnalytics({
+      ...uploadAnalytics,
+      chunks,
+    });
+  };
+
+  const handleChunkSuccess: MuxUploaderProps['onChunkSuccess'] = ({ detail }) => {
+    const chunks = [...uploadAnalytics.chunks];
+    chunks[detail.chunk].uploadFinished = Date.now();
+
+    setUploadAnalytics({
+      ...uploadAnalytics,
+      chunks,
+    });
+  };
+
+  const handleSuccess: MuxUploaderProps['onSuccess'] = () => {
+    const finalAnalytics = {...uploadAnalytics};
+    finalAnalytics.uploadFinished = Date.now();
+
+    fetch('/api/telemetry', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        type: 'upload',
+        data: finalAnalytics,
+      })
+    });
+
+    setIsPreparing(true);
+  };
+
+  useEffect(() => {
+    if (upload && upload.asset_id) {
+      Router.push({
+        pathname: `/assets/${upload.asset_id}`,
+      });
+    }
+  }, [upload]);
+
+  if (errorMessage) {
+    return (
+      <Layout>
+        <div style={{ paddingBottom: '20px'}}><h1>{errorMessage}</h1></div>
+        <div>
+          <Button onClick={Router.reload}>Reset</Button>
+        </div>
+      </Layout>
+    );
   }
 
   return (
     <Layout
-      onFileDrop={onDrop}
+      dragActive
+      isUploading={isUploading}
     >
-      <div>
-        <div>
-          <h1>Add a video.</h1>
-          <h1>Get a shareable link to stream it.</h1>
-        </div>
-        <div className="cta">
-          <div className="drop-notice">
-            <h2>↓ Drag & drop a video file anywhere</h2>
+      <div style={{ width: isUploading ? '100%' : 'auto' }}>
+        {!isUploading ? (
+          <div>
+            <h1>Add a video.</h1>
+            <h1>Get a shareable link to stream it.</h1>
           </div>
-          <label htmlFor="file-input">
-            <Button type="button" onClick={() => inputRef.current && inputRef.current.click()}>
-              <span className="cta-text-mobile">Add a video</span>
-              <span className="cta-text-desktop">Upload a video</span>
-            </Button>
-            <input id="file-input" type="file" onChange={onInputChange} ref={inputRef} />
-          </label>
-          <div className="cta-record">
-            <Link href="/record?source=camera"><Button>Record from camera</Button></Link>
-          </div>
-          <div className="cta-record">
-            <Link href="/record?source=screen"><Button>Record my screen</Button></Link>
+          ) : null}
+        <div className={isUploading ? '' : 'cta'}>
+          {!isUploading ? (
+            <div className="drop-notice">
+              <h2>↓ Drag & drop a video file anywhere</h2>
           </div>
+          ) : null}
+          <MuxUploader 
+            onUploadStart={handleUpload}
+            onChunkAttempt={handleChunkAttempt}
+            onChunkSuccess={handleChunkSuccess}
+            onSuccess={handleSuccess}
+            style={{ 
```

Comment:
suggestion(non-blocking): Either now or as a followup, we should try to move these inline styles to the style blocks for consistency (and validation that we don't have any issues there)